pkg/framework/provider/crypto/crypto.go

```diff
@@ -0,0 +1,140 @@
+// and kms API to be used everywhere by the framework. No call should be directly made to the below
+// implementation. Instead use pkg/common/crypto to use these operations.
+
+// TinkCrypto is the default Crypto SPI implementation using Tink
+type TinkCrypto struct {
+}
+
+// Encrypt will encrypt msg using the implementation's corresponding encryption key and primitive in kh
+func (t *TinkCrypto) Encrypt(msg, aad []byte, kh interface{}) ([]byte, []byte, error) {
+	keyHandle, ok := kh.(*keyset.Handle)
+	if !ok {
+		return nil, nil, errors.New("bad key handle format")
+	}
+
+	a, err := aead.New(keyHandle)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ct, err := a.Encrypt(msg, aad)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ps, err := keyHandle.Primitives()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ivSize := nonceSize(ps)
+	prefixLength := len(ps.Primary.Prefix)
+	cipherText := ct[prefixLength+ivSize:]
+	nonce := ct[prefixLength : prefixLength+ivSize]
+
+	return cipherText, nonce, nil
+}
+
+func nonceSize(ps *primitiveset.PrimitiveSet) int {
+	ivSize := aeadsubtle.AESGCMIVSize
+	// AESGCM and XChacha20Poly1305 nonce sizes supported only for now
+	if _, ok := (ps.Primary.Primitive).(*aeadsubtle.XChaCha20Poly1305); ok {
+		ivSize = chacha20poly1305.NonceSizeX
+	}
+
+	return ivSize
+}
+
+// Decrypt will decrypt cipher using the implementation's corresponding encryption key
+func (t *TinkCrypto) Decrypt(cipher, nonce, aad []byte, kh interface{}) ([]byte, error) {
+	keyHandle, ok := kh.(*keyset.Handle)
+	if !ok {
+		return nil, errors.New("bad key handle format")
+	}
+
+	a, err := aead.New(keyHandle)
+	if err != nil {
+		return nil, err
+	}
+
+	ps, err := keyHandle.Primitives()
+	if err != nil {
+		return nil, err
+	}
+
+	ct := make([]byte, 0, len(ps.Primary.Prefix)+len(nonce)+len(cipher))
+	ct = append(ct, ps.Primary.Prefix...)
+	ct = append(ct, nonce...)
+	ct = append(ct, cipher...)
+
+	pt, err := a.Decrypt(ct, aad)
+	if err != nil {
+		return nil, err
+	}
+
+	return pt, nil
+}
+
+// Sign will sign msg using the implementation's corresponding signing key
+func (t *TinkCrypto) Sign(msg []byte, kh interface{}) ([]byte, error) {
+	keyHandle, ok := kh.(*keyset.Handle)
+	if !ok {
+		return nil, errors.New("bad key handle format")
+	}
+
+	signer, err := signature.NewSigner(keyHandle)
+	if err != nil {
+		return nil, err
+	}
+
+	s, err := signer.Sign(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+// Verify will verify sig signature of msg using the implementation's corresponding signing key
+func (t *TinkCrypto) Verify(sig, msg []byte, kh interface{}) error {
+	keyHandle, ok := kh.(*keyset.Handle)
+	if !ok {
+		return errors.New("bad key handle format")
+	}
+
+	verifier, err := signature.NewVerifier(keyHandle)
+	if err != nil {
+		return err
+	}
+
+	err = verifier.Verify(sig, msg)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
`return verifier.Verify(sig, msg)` or wrap the error

Reply:
done